src/Symfony/Component/Profiler/Profiler.php

```diff
@@ -0,0 +1,387 @@
+        }
+
+        return $ret;
+    }
+
+    /**
+     * Purges all data from the storage.
+     */
+    public function purge()
+    {
+        $this->storage->purge();
+    }
+
+    /**
+     * Exports the current profiler data.
+     *
+     * @param ProfileInterface $profile A Profile instance
+     *
+     * @return string The exported data
+     */
+    public function export(ProfileInterface $profile)
+    {
+        return base64_encode(serialize($profile));
+    }
+
+    /**
+     * Imports data into the profiler storage.
+     *
+     * @param string $data A data string as exported by the export() method
+     *
+     * @return ProfileInterface A Profile instance
+     */
+    public function import($data)
+    {
+        $profile = unserialize(base64_decode($data));
+
+        if ($this->storage->read($profile->getToken())) {
+            return false;
+        }
+
+        $this->save($profile);
+
+        return $profile;
+    }
+
+    /**
+     * Finds profiler tokens for the given criteria.
+     *
+     * @param array  $criteria The Criteria
+     * @param string $limit    The maximum number of tokens to return
+     * @param string $start    The start date to search from
+     * @param string $end      The end date to search to
+     *
+     * @return array An array of tokens
+     *
+     * @see http://php.net/manual/en/datetime.formats.php for the supported date/time formats
+     */
+    public function findBy(array $criteria, $limit, $start, $end)
+    {
+        return $this->storage->findBy($criteria, $limit, $this->getTimestamp($start), $this->getTimestamp($end));
+    }
+
+    /**
+     * Finds profiler tokens for the given criteria.
+     *
+     * @param string $ip     The IP
+     * @param string $url    The URL
+     * @param string $limit  The maximum number of tokens to return
+     * @param string $method The request method
+     * @param string $start  The start date to search from
+     * @param string $end    The end date to search to
+     *
+     * @return array An array of tokens
+     *
+     * @see http://php.net/manual/en/datetime.formats.php for the supported date/time formats
+     * @deprecated since 2.8 and will be removed in 3.0 use Profiler::findBy instead
+     */
+    public function find($ip, $url, $limit, $method, $start, $end)
+    {
+        $criteria = array();
+        if (!empty($ip)) {
+            $criteria['ip'] = $ip;
+        }
+        if (!empty($url)) {
+            $criteria['url'] = $url;
+        }
+        if (!empty($method)) {
+            $criteria['method'] = $method;
+        }
+
+        return $this->findBy($criteria, $limit, $start, $end);
+    }
+
+    /**
+     * Profile a Request.
+     *
+     * @param Request  $request
+     * @param Response $response
+     *
+     * @return ProfileInterface|void
+     */
+    public function profileRequest(Request $request, Response $response)
```

Comment:
I don't like this method because it's coupled to HttpFoundation. I think it should be removed. Setting the "X-Debug-Token" header is not the responsibility of a profiler.